app/scripts/components/dataset-location-filter/dataset-location-filter-actions.js

```diff
@@ -0,0 +1,11 @@
+import { createAction, createThunkAction } from 'redux-tools';
+
+import { updateURL } from 'actions/links';
+
+export const setLocation = createAction('datasets/setLocation');
+export const setDatasetsLocation = createThunkAction('datasets/onSetDatasetsLocation', location =>
+  (dispatch) => {
+    dispatch(setLocation(location));
+    dispatch(updateURL());
```

Comment:
I don't think that this would be re-usable as long as you are going to always update the URL... Which one? always the explore one? What if you need to update another URL? What if you don't need to update an URL?